### DISCUSSION TITLE
How to specify a branch for pull?

### DISCUSSION THREAD

**Discussion by harsiddhdave44** [gitpython-developers/GitPython#1471] [Q&A] [answered]
I've been looking for the Python version of the following git command:
`git pull origin master`
So far, I've tried to inspect the code to see if there's a way to specify a branch for the pull, but I couldn't find any. I guess I'd have to pass the branch in `kwargs` of `repo_obj.remotes.origin.pull`, but I'm not sure if it's the right way to do it.

Could someone please explain to me why the pull function doesn't have any parameter for this, and how can I specify the branch for the pull?

**Comment by Byron** [comment, marked as answer]
It's possible to configure the _refspec_ as first argument to `pull` (as documented in the [fetch method](https://gitpython.readthedocs.io/en/stable/reference.html?highlight=Remote#git.remote.Remote.fetch)) can be `refs/heads/local-branch:refs/heads/upstream-branch` to fetch `upstream-branch` and bring it into `local-branch`.
